src/get_context_sequences.py

```diff
@@ -0,0 +1,215 @@
+    
+    prev_idx = False
+    elem_idx = ''
+    prev_sequence = ''
+    next_sequence = ''
+    prev_elem_sequence = ''
+    for tag, elem in elem_iter(root):
+        if tag == 'note':
+            elem_sequence = elem.text
+            elem_idx = elem.attrib['{http://www.w3.org/XML/1998/namespace}id']
+            
+            if prev_idx == True:
+                next_sequence = elem_sequence
+                context_sequence = concat_sequences_full(prev_sequence, curr_sequence, next_sequence, max_length)
+                
+                id_list.append(idx)
+                context_sequence_list.append(context_sequence)
+            
+
+            idx = elem_idx
+            curr_sequence = elem_sequence
+            prev_sequence = prev_elem_sequence
+                
+            prev_elem_sequence = elem_sequence
+            prev_idx = True
+        elif tag == 'u':
+            for child in elem.getchildren():
+                elem_sequence = child.text
+                elem_idx = child.values()[0]
+                
+                if prev_idx == True:
+                    next_sequence = elem_sequence
+                    context_sequence = concat_sequences_full(prev_sequence, curr_sequence, next_sequence, max_length)
+                    
+                    id_list.append(idx)
+                    context_sequence_list.append(context_sequence)
+                    
+                
+                idx = elem_idx
+                curr_sequence = elem_sequence
+                prev_sequence = prev_elem_sequence
+                    
+                prev_elem_sequence = elem_sequence
+                prev_idx = True
+                
+    next_sequence = ''
+    context_sequence = concat_sequences_full(prev_sequence, curr_sequence, next_sequence, max_length)
+    
+    id_list.append(idx)
+    context_sequence_list.append(context_sequence)
+    
+    
+    output_dict = {'id' : id_list,
+                   'context_sequence' : context_sequence_list}
+    return output_dict
+
+def main(args):
+    
+    # chose type of context sequence to output
+    if args.context_type == 'left_context':
+        context_seq_func = get_context_sequence_left
+    elif args.context_type == 'full_context':
+        context_seq_func = get_context_sequence_full
+    else:
+        raise Exception(f'{args.context_type} is an invalid context type. Valid inputs are: [left_context, full_context]')
+    
+    protocols = sorted(list(protocol_iterators(args.records_folder, start=args.start, end=args.end)))
+    curr_year = protocols[0].split('\\')[-2]
+    
+    context_sequence_dict = {'id' : [],
+                             'context_sequence' : []}
+    
+    for protocol in progressbar.progressbar(protocols):
+        next_year = protocol.split('\\')[-2]
+        if curr_year != next_year:
+            output_df = pd.DataFrame.from_dict(context_sequence_dict)
+            save_file = curr_year + '_' + args.context_type + '.csv'
```

Comment:
We want to avoid saving to a CSV altogether, but that's a worry for another day.